pkg/operation/botanist/component/kubecontrollermanager/kube_controller_manager.go

```diff
@@ -290,27 +308,89 @@ func (k *kubeControllerManager) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	if _, err := controllerutil.CreateOrUpdate(ctx, k.seedClient, vpa, func() error {
-		vpa.Spec.TargetRef = &autoscalingv1.CrossVersionObjectReference{
-			APIVersion: appsv1.SchemeGroupVersion.String(),
-			Kind:       "Deployment",
-			Name:       v1beta1constants.DeploymentNameKubeControllerManager,
+	if k.hvpaConfig != nil && k.hvpaConfig.Enabled {
+		if err := kutil.DeleteObject(ctx, k.seedClient, vpa); err != nil {
+			return err
 		}
-		vpa.Spec.UpdatePolicy = &autoscalingv1beta2.PodUpdatePolicy{
-			UpdateMode: &vpaUpdateMode,
+
+		var (
+			updateModeAuto = hvpav1alpha1.UpdateModeAuto
+			vpaLabels      = map[string]string{v1beta1constants.LabelRole: "kube-controller-manager-vpa"}
+		)
+
+		scaleDownUpdateMode := k.hvpaConfig.ScaleDownUpdateMode
+		if scaleDownUpdateMode == nil {
+			scaleDownUpdateMode = pointer.StringPtr(hvpav1alpha1.UpdateModeAuto)
 		}
-		vpa.Spec.ResourcePolicy = &autoscalingv1beta2.PodResourcePolicy{
-			ContainerPolicies: []autoscalingv1beta2.ContainerResourcePolicy{{
-				ContainerName: containerName,
-				MinAllowed: corev1.ResourceList{
-					corev1.ResourceCPU:    resource.MustParse("100m"),
-					corev1.ResourceMemory: resource.MustParse("100Mi"),
+
+		if _, err := controllerutil.CreateOrUpdate(ctx, k.seedClient, hvpa, func() error {
+			hvpa.Labels = getLabels()
+			hvpa.Spec.Replicas = pointer.Int32Ptr(1)
+			hvpa.Spec.Hpa = hvpav1alpha1.HpaSpec{
+				Deploy:   false,
+				Selector: &metav1.LabelSelector{MatchLabels: getLabels()},
+				Template: hvpav1alpha1.HpaTemplate{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: getLabels(),
+					},
+					Spec: hvpav1alpha1.HpaTemplateSpec{
+						MinReplicas: pointer.Int32Ptr(int32(1)),
+						MaxReplicas: int32(1),
+					},
 				},
-			}},
+			}
+			hvpa.Spec.Vpa = hvpav1alpha1.VpaSpec{
+				Selector: &metav1.LabelSelector{MatchLabels: vpaLabels},
+				Deploy:   true,
+				ScaleUp: hvpav1alpha1.ScaleType{
+					UpdatePolicy: hvpav1alpha1.UpdatePolicy{
+						UpdateMode: &updateModeAuto,
+					},
+				},
+				ScaleDown: hvpav1alpha1.ScaleType{
+					UpdatePolicy: hvpav1alpha1.UpdatePolicy{
+						UpdateMode: scaleDownUpdateMode,
+					},
+				},
+				Template: hvpav1alpha1.VpaTemplate{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: vpaLabels,
+					},
+					Spec: hvpav1alpha1.VpaTemplateSpec{
+						ResourcePolicy: vpaResourcePolicy,
+					},
+				},
+			}
+			hvpa.Spec.TargetRef = &autoscalingv2beta1.CrossVersionObjectReference{
+				APIVersion: appsv1.SchemeGroupVersion.String(),
+				Kind:       "Deployment",
+				Name:       v1beta1constants.DeploymentNameKubeControllerManager,
+			}
+			return nil
```

Comment:
I think the `WeightBasedScalingIntervals` should be defined as done for ETCD [here](https://github.com/rfranzke/gardener/blob/81f939bcfceae96bf96f5ef366bffcd5373e43d2/pkg/operation/botanist/component/etcd/etcd.go#L479-L485).

Reply:
What is this/why do I need to care? I'll add it, though, I'm only interested in using HVPA to get a VPA to be able to disable scale-down, that's it. I'm not interested in any extra configuration possible with HVPA - I just want the default behaviour.

Reply:
It is not at all clear to me what is the _default behaviour_ for HVPA should be. From your perspective, I see that you expect that VPA behaviour should be default. It is not at all clear to me why should that be 🤷

Reply:
For me, HVPA allows to combine both HPA and VPA with enhanced scaling strategies and policies. I think most of the "special" configurations should be entirely optional. Also, it should be possible to only use HPA (but not VPA) and vice versa. Yes, you can set `deploy=false`, but today, you still have to provide some dummy values leading to confusion for end-users not familiar with the HVPA controller internals.
Anyways, now we've sorted it out and all of this is an alpha feature with a hopefully short lifetime.

Reply:
With v2, some of the configuration burden (and limitations like not being able to control horizontal scaling too much) will go away. There is still some cleanup of the API required, unfortunately. I will check if someone can work on the clean up as well the integration.